File: 0101_0200/0131/0131.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    a = k = -1
    for i in range(n):
        t, s = (int(x) for x in f.readline().split())
        if s and t > k:
            k = t
            a = i + 1
    q.write(str(a))

File: 0101_0200/0131/0131.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, t, s, k = -1, a = -1;
	fscanf(f, "%d", &n);
	for (int i = 1; i <= n; ++i)
	{
		fscanf(f, "%d%d", &t, &s);
		if (s && t > k)
			k = t, a = i;
	}
	fprintf(q, "%d", a);
	return 0;
}
